sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -48,7 +48,7 @@ class InMemoryCatalog extends ExternalCatalog {
   private val catalog = new scala.collection.mutable.HashMap[String, DatabaseDesc]
 
   private def filterPattern(names: Seq[String], pattern: String): Seq[String] = {
-    val regex = pattern.replaceAll("\\*", ".*").r
+    val regex = pattern.replaceAll("\\.\\*", "\\*").replaceAll("\\*", "\\.\\*").r
```

Comment:
I did a deep investigation of pattern matching for LIKE in show tables/functions. Here is what I found: Hive only supports \* and | as wildcards. Use ".\*" to replace "\*" is right. The only issue is ShowFunctions() in commands.scala did not use it, thus cause the test cases fail. By using listFunctions() in SessionCatalog, the problem should be resolved. Thanks.